### PR TITLE
refactor: convert OnRenderProcessGoneDelegateTest to Kotlin

### PR DESCRIPTION
**Pull request by david-allison**
## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/cardviewer/OnRenderProcessGoneDelegateTest.kt [hunk 1]
```diff
@@ -13,213 +13,180 @@
...
-    }
-
+    fun unrecoverableCrashCloses() {
+        val mock = minimisedViewer
+        val delegate = getInstance(mock)
 
-    protected void callOnRenderProcessGone(OnRenderProcessGoneDelegateImpl delegate) {
-        callOnRenderProcessGone(delegate, delegate.getTarget().getWebView());
-    }
+        doReturn(null).`when`(mock).currentCard
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/cardviewer/OnRenderProcessGoneDelegateTest.kt:119]: kotlincleanup when / is

File: AnkiDroid/src/test/java/com/ichi2/anki/cardviewer/OnRenderProcessGoneDelegateTest.kt [hunk 2]
```diff
@@ -13,213 +13,180 @@
...
-        return ret;
+    protected val minimisedViewer: AbstractFlashcardViewer
+        get() = getViewer(Lifecycle.State.CREATED)
+    protected val viewer: AbstractFlashcardViewer
+        get() = getViewer(Lifecycle.State.STARTED)
+
+    private fun getViewer(state: Lifecycle.State): AbstractFlashcardViewer {
+        val mockWebView = mock(WebView::class.java)
+        val mock = strictMock(AbstractFlashcardViewer::class.java)
+        doReturn(mock(Lock::class.java)).`when`(mock).writeLock
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/anki/cardviewer/OnRenderProcessGoneDelegateTest.kt:146]: cleanup when / whenever

**Review by mikehardy — CHANGES_REQUESTED** [review]
one maybe on nullability, mostly just when/is

**Comment by david-allison** [comment]
I've moved when/is to annotations on the class

**Review by mikehardy — APPROVED** [review]
🚀

**Comment by mikehardy** [comment]
'DeckPickerTest > version16CollectionOpens' is flaky on windows

Seen in another instance or two as well? Will re-run failed unit test once it's possible (other parts of OS matrix have to finish first)

```

com.ichi2.anki.DeckPickerTest > checkDisplayOfStudyOptionsOnTablet[0] SKIPPED
thread '<unnamed>' panicked at 'Anki already open, or media currently syncing.', src/lib.rs:251:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'DBError { info: "SqliteFailure(Error { code: Unknown, extended_code: 1 }, Some(\"no such table: decks\"))", kind: Other }', src/lib.rs:256:13
thread '<unnamed>' panicked at 'Anki already open, or media currently syncing.', src/lib.rs:251:25

com.ichi2.anki.DeckPickerTest > version16CollectionOpens[1] FAILED
    java.lang.AssertionError: Collection should now be open
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:26)
        at com.ichi2.anki.DeckPickerTest.version16CollectionOpens(DeckPickerTest.java:411)

```

**Pull request merged**